tools/toolchain/optimized_clang.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash -uex
+
+if [[ -z "${CLANG_BUILD}" ]]; then
+    echo "CLANG_BUILD not specified"
+    exit 1
+elif [[ "${CLANG_BUILD}" = "SKIP" ]]; then
+    echo "Skip building optimized clang"
+    exit 0
+elif [[ "${CLANG_BUILD}" = "INSTALL" ]]; then
+    echo "build and install optimized clang"
+elif [[ "${CLANG_BUILD}" = "INSTALL_FROM" ]]; then
+    echo "install prebuilt optimized clang"
+else
+    echo "Not sure what to do with ${CLANG_BUILD}"
+    exit 1
+fi
+
+if [[ "${CLANG_BUILD}" = "INSTALL" ]] || [[ "${CLANG_BUILD}" = "INSTALL_FROM" ]]; then
+    if [[ -z "${CLANG_ARCHIVE}" ]]; then
+        echo "CLANG_ARCHIVE does not specified"
+        exit 1
+    fi
+fi
+
+ARCH="$(arch)"
+LLVM_EXPERIMENTAL_TARGETS_TO_BUILD="WebAssembly"
+if [[ "${ARCH}" = "x86_64" ]]; then
+    LLVM_TARGET_ARCH=X86
+elif [[ "${ARCH}" = "aarch64" ]]; then
+    LLVM_TARGET_ARCH=AArch64
+    LLVM_EXPERIMENTAL_TARGETS_TO_BUILD="${LLVM_EXPERIMENTAL_TARGETS_TO_BUILD};AArch64"
```

Comment:
now that we are using the bash dialect, in future, could use
```sh
LLVM_EXPERIMENTAL_TARGETS_TO_BUILD+=";AArch64"
```

Reply:
also, `WebAssembly` duplicates the one specified in `-DLLVM_TARGETS_TO_BUILD` option added to "CLANG_OPTS". it does not hurt. but it could be confusing.